src/chart/bar/PictorialBarView.js

```diff
@@ -227,7 +227,9 @@ function prepareBarLength(itemModel, symbolRepeat, layout, opt, output) {
         output.repeatCutLength = layout[valueDim.wh];
     }
 
-    output.pxSign = boundingLength > 0 ? 1 : boundingLength < 0 ? -1 : 0;
+    // if 'pxSign' means sign of pixel,  it can't be zero, or symboScale will be zero
+    // and when borderwidth be setted, the actual linewidth will be NaN
+    output.pxSign = boundingLength > 0 ? 1 : -1;
```

Comment:
I think no change to be done here. It is just used to indicate the sign of pixel.

https://github.com/apache/incubator-echarts/blob/3ab134179298a780659be9009bce2c0f7ec6bec9/src/chart/bar/PictorialBarView.js#L296

The reason of the bug is beacuse `pathForLineWidth.getLineScale()` may be zero, as we all know, divisor can not be `0`, which is meaningless. So there should be a condition to judge whether `lineScale` is `0` or not before
```js
valueLineWidth /= pathForLineWidth.getLineScale();
valueLineWidth *= symbolScale[opt.valueDim.index];
```
if `0`, don't perform the following operations and set `valueLineWidth` to `0` directly.
Maybe changes should be like below:
```js
var lineScale = pathForLineWidth.getLineScale();
if (lineScale) {
    valueLineWidth /= lineScale;
    valueLineWidth *= symbolScale[opt.valueDim.index];
}
else {
    valueLineWidth = 0;
}
```

Reply:
I originally planned to modify it like this, but if it is modified like this, the graph representing this data will disappear. When symbolSize is specified, should such a display be made?
And as a pixel sign, why can the sign be 0?
Based on the above doubts, I made the current modification. Under this modification, when the symbolSize is specified, even if the data is 0, the data will be displayed at the corresponding data position.
However, I also think that a conditional selection statement should also be added here to prevent the situation where the dividend is 0.

Reply:
I originally planned to modify it like this, but if it is modified like this, the graph representing this data will disappear. When symbolSize is specified, should such a display be made?
And as a pixel sign, why can the sign be 0?
Based on the above doubts, I made the current modification. Under this modification, when the symbolSize is specified, even if the data is 0, the data will be displayed at the corresponding data position.
However, I also think that a conditional selection statement should also be added here to prevent the situation where the dividend is 0.

Sorry, I just used the wrong account, so delete the previous comment and re-send it.🌚

Reply:
@yanheSu Now I think you are right, maybe the sign of pixel shouldn't be `0` but negative or positive. And as you said, if `symbolSize` is specified, the chart also should draw it even though its data is `0`.

Reply:
@plainheart 
From my personal point of view, I don’t think this is very strange. When `symboSize` is specified and `symbolRepeat` is not turned on, it is not a problem to draw a symbol of size `symbolSize` at the position of `symbolPosition`.
0 value is just a special point, all `symbolPosition` positions coincide at one point.